File: data-structures/linked_list.cpp
#include<bits/stdc++.h>
using namespace std;

class node{
  public:
    int data;
    node *next;

    node(int x){
      data = x;
      next = NULL;
    }
};

node *head = NULL;

void insert_front(int data){
  node *temp = new node(data);
  temp->next = head;
  head = temp;
}

void insert_back(int data){
  node *temp = new node(data);

  node *t = head;
  while(t->next) t = t->next;
  t->next = temp;
}

void delete_node(int data){
  node *temp = head;
  if(temp->data == data){
    head = temp->next;
    free(temp);
  }
  
  while(temp->next){
    if(temp->next->data == data){
      node *curr = temp->next;
      temp->next = curr->next;
      free(curr);
    }
    temp = temp->next;
  }
}

void print(){
  node *temp = head;
  while(temp){
    cout << temp->data <<" ";
    temp = temp->next;
  }
}

int main(){
  for (int i = 10; i > 0; i--) insert_front(i);
  for (int i = 11; i <= 20; i++) insert_back(i);
  delete_node(9);
  print();
}

File: data-structures/linked_list.py
class Node:
  def __init__(self, key):
    self.key = key
    self.next = None 

class LinkedList:
  def __init__(self):
    self.head = None

  def print(self):
    temp = self.head
    while temp:
      print(temp.key, end=" ")
      temp = temp.next
    print()

  def append(self, key):
    node = Node(key)
    if not self.head: self.head = node
    else:
      temp = self.head
      while temp.next: temp = temp.next
      temp.next = node
  
  def prepend(self, key):
    node = Node(key)
    node.next = self.head
    self.head = node
  
  def delete(self, key):
    prev = self.head
    temp = prev.next
    
    if prev.key == key:
      del(prev)
      self.head = temp
      prev = temp
      temp = temp.next

    while temp:
      if temp.key == key:
        prev.next = temp.next
        del(temp)
        temp = prev
      prev = temp
      temp = temp.next

linked_list = LinkedList()

for i in range(5):
  linked_list.append(i)

for i in range(5, 10):
  linked_list.prepend(i)

linked_list.print()
linked_list.delete(7)
linked_list.print()